fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -466,6 +473,113 @@ where
     }
 }
 
+impl<S, Tx, Ecal> Interpreter<S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    fn upgrade_inner(
+        upgrade: &mut Upgrade,
+        storage: &mut S,
+        initial_balances: InitialBalances,
+        gas_costs: &GasCosts,
+        fee_params: &FeeParameters,
+        base_asset_id: &AssetId,
+        gas_price: Word,
+    ) -> Result<(), InterpreterError<S::DataError>> {
+        let metadata = upgrade.metadata().as_ref();
+        debug_assert!(
+            metadata.is_some(),
+            "`upgrade_inner` is called without cached metadata"
+        );
+
+        match upgrade.upgrade_purpose() {
+            UpgradePurpose::ConsensusParameters { .. } => {
+                let consensus_parameters = if let Some(metadata) = metadata {
+                    Self::get_consensus_parameters(&metadata.body)?
+                } else {
+                    let metadata = UpgradeMetadata::compute(upgrade)?;
+                    Self::get_consensus_parameters(&metadata)?
+                };
+
+                let current_version = storage
+                    .consensus_parameters_version()
+                    .map_err(RuntimeError::Storage)?;
+                let next_version = current_version
+                    .checked_add(1)
+                    .ok_or(InterpreterError::Panic(PanicReason::ArithmeticOverflow))?;
```

Comment:
Introducing a new `PanicReason` for this would be nice. `ArithmeticOverflow` is meant to be used for arithmetic opcodes only.